ec2-snapshot.sh

```diff
@@ -31,35 +42,19 @@ do
   # indirection in bash, the best we can do is check for empty string (which
   # shouldn't be a problem in this case as an empty string can't be used with
   # the aws cli)
-  eval check=\$$v
-  if [ -z "${check}" ]; then
+  if [[ -z ${!v} ]]; then
     die "env var $v is required"
   fi
 done
 
-# install the awscli util
-if [[ ! -e venv/bin/activate ]]; then
-    virtualenv venv
-fi
-
-. venv/bin/activate
-
-pip install -q awscli
-
-# install ec2-automate-backup-awscli.sh
-# per https://github.com/colinbjohnson/aws-missing-tools/issues/106
-# this script was renamed to ec2-automate-backup.sh
-# XXX pinning to the commit sha
-BACKUP_SCRIPT="ec2-automate-backup.sh"
-wget --no-clobber --no-verbose "https://raw.githubusercontent.com/colinbjohnson/aws-missing-tools/1b6cd230dde529f3bf4c19ea80fccdf42e479dae/ec2-automate-backup/${BACKUP_SCRIPT}"
-chmod a+x "$BACKUP_SCRIPT"
-
 # lookup volume-ids for our instance-id; assuming only one volume is mounted
-VOLUME_ID="$(aws ec2 describe-volumes --region "$REGION" --filters Name=attachment.instance-id,Values="${INSTANCE_ID}" | ./jq --raw-output '.Volumes[0].VolumeId')"
+VOLUME_ID="$(aws ec2 describe-volumes --region "$REGION" --filters Name=attachment.instance-id,Values="${INSTANCE_ID}" | jq --raw-output '.Volumes[0].VolumeId')"
 
 # option snapshot our volume-id
 
 # XXX for unknown reasons, ec2-automate-backup.sh defaults to EC2_REGION
 # instead of AWS_DEFAULT_REGION -- so we are setting it an exclitly as a cli
 # option
-"./${BACKUP_SCRIPT}" -v "$VOLUME_ID" -r "$REGION" -k 91d -n -p
+"$BACKUP_SCRIPT" -v "$VOLUME_ID" -r "$REGION" -k 91d -n -p
+
+# vim: tabstop=2 shiftwidth=2 expandtab
```

Comment:
Ugh.  Not my editor of choice.  Do we have to have this in the file?

Reply:
Yes. 
https://trends.google.com/trends/explore?q=vim,emacs